### ISSUE TITLE
Create configuration that specifies what bundles are needed to load

### ISSUE THREAD

**Issue by matthewp** [stealjs/steal-tools#541]
If you provide steal-tools something like:

```js
stealTools.build({
  bundle: [
    "home",
    "register",
    "product"
  ]
});
```

This could result in shared bundles being created. Ideally you want to prevent a waterfall scenario, which you could do by listing the scripts in your html:

```js
<script src="/dist/bundles/app/main.js" defer></script>
<script src="/dist/bundles/app/home.js" defer></script>
<script src="/dist/bundles/app/home-register.js" defer></script>
<script src="/dist/bundles/app/home-register-product.js" defer></script>
```

The browser will fetch all of these scripts in parallel so there is no waterfall.

The **_problem_** is that a user doesn't know what script tags they need to add. What I am proposing is that we write out this information somewhere that you can use in your server-side templating. 

One idea I had (not necessarily the best) is to write a json file into your dest, maybe `dist/bundles.json` that looks like:

```json
{
  "bundles/app/home": {
    "main": "bundles/app/main",
    "shared": [
      "bundles/app/home-register",
      "bundles/app/home-register-product"
    ]
  }
}
```

**Comment by matthewp** [comment]
@m-mujica This is the next thing to do for slim.  I think we just want a config option to turn this on for now.  Maybe `bundleManifest: true`? I think we should use this format: https://github.com/GoogleChrome/http2-push-manifest . as it could be useful to add stuff to it in the future.

**Comment by matthewp** [comment]
Arg, that format assumes you know the routes. steal-tools is not route-aware, unfortunately.  I guess we could still do the same looking format, but instead of the keys being routes, the keys are the bundle names.

**Comment by matthewp** [comment]
Closed by #741
